dbt/adapters/bigquery/impl.py

```diff
@@ -47,11 +47,16 @@ class PartitionConfig(dbtClassMixin):
     data_type: str = "date"
     granularity: str = "day"
     range: Optional[Dict[str, Any]] = None
+    time_ingestion_partitioning: bool = False
+    copy_partitions: bool = False
```

Comment:
Is there a use case for copy_partitions outside of time_ingestion_partitioning or for not using copy_partitions with time_ingestion_partitioning? Just wondering if we need another config parameter

Reply:
Yes, absolutely, it's also much faster than a merge so I'm going to use it on all my models whether it's time ingestion partitioning or column type partitioning! It really fits my use case because I'm usually processing models partition per partition but it might not fit everyone (though honestly, most people in my opinion that can afford a temporary loss of consistency for a speed gain 👍 )

Reply:
Is there a use case for `time_ingestion_partitioning = True` and `copy_partitions = False `? Should we allow users to do this?

@Fleid - I know you had concerns with adding more variables for users to set but I think we need to keep these separate

Reply:
Yes, here some use cases:
`time_ingestion_partitioning = True` and `copy_partitions = False` => for atomic merge for time ingestion  partitioned table (useful for people that can't afford potential inconsistency)
`time_ingestion_partitioning = False` and `copy_partitions = True` => for fast atomic insert overwrite on a single partition or fast non atomic on multiple partitions (quite common at my company)
`time_ingestion_partitioning = False` and `copy_partitions = False` => before the 2 PRs
`time_ingestion_partitioning = True` and `copy_partitions = True` => for fast atomic insert overwrite on a single partition or fast non atomic on multiple partitions (a common use case at my company)

Reply:
Great context! I'm going to do a follow up PR after we merge to update our documentation with this context.